File: opt/B-Orthogonality.py
N = int(input())
A = list(map(int,input().split()))
B = list(map(int,input().split()))

inner_product = 0
for i in range(N):
    inner_product += A[i] * B[i]

if inner_product == 0:
    ans = 'Yes'
else:
    ans = 'No'

print(ans)

File: B-Orthogonality.cpp
#include <iostream>
#include <vector>
#include <string>
using namespace std;

int main(){
    int n;
    cin >> n;
    vector<int> A(n);
    vector<int> B(n);
    for (int i = 0; i< n;i++) cin >> A.at(i);
    for (int i = 0; i < n; i ++) cin >> B.at(i);

    int inner_product = 0;
    for (int i = 0; i < n; i ++){
        inner_product += A.at(i) * B.at(i);

    }
    if (inner_product == 0){
        cout << "Yes" << endl;
    }else{
        cout << "No" << endl;
    }

}